feature_engine/selection/recursive_feature_addition.py

```diff
@@ -0,0 +1,269 @@
+        feature_importances_cv.index = self.variables
+
+        # Aggregate the feature importance returned in each fold
+        self.feature_importances_ = feature_importances_cv.mean(axis=1)
+
+        # Sort the feature importance values descreasingly
+        self.feature_importances_.sort_values(ascending=False, inplace=True)
+
+        # Extract most important feature from the ordered list of features
+        first_most_important_feature = list(self.feature_importances_.index)[0]
+
+        # Run baseline model using only the most important feature
+        baseline_model = cross_validate(
+            self.estimator,
+            X[first_most_important_feature].to_frame(),
+            y,
+            cv=self.cv,
+            scoring=self.scoring,
+            return_estimator=True,
+        )
+
+        # Save baseline model performance
+        baseline_model_performance = baseline_model["test_score"].mean()
+
+        # list to collect selected features
+        # It is initialized with the most important feature
+        self.selected_features_ = [first_most_important_feature]
+
+        # dict to collect features and their performance_drift
+        # It is initialized with the performance drift of
+        # the most important feature
+        self.performance_drifts_ = {
+            first_most_important_feature: baseline_model_performance -
+            self.initial_model_performance_
+        }
+
+        # loop over the ordered list of features by feature importance starting
+        # from the second element in the list.
+        # Train the estimator with
+        for feature in list(self.feature_importances_.index)[1:]:
+
+            # Add feature and train new model
+            model_tmp = cross_validate(
+                self.estimator,
+                X[self.selected_features_ + [feature]],
+                y,
+                cv=self.cv,
+                scoring=self.scoring,
+                return_estimator=True,
+            )
+
+            # assign new model performance
+            model_tmp_performance = model_tmp["test_score"].mean()
+
+            # Calculate performance drift
+            performance_drift = model_tmp_performance - baseline_model_performance
+
+            # Save feature and performance drift
+            self.performance_drifts_[feature] = performance_drift
+
+            # If new performance model is
+            if performance_drift > self.threshold:
+
+                # add feature to the list of selected features
+                self.selected_features_.append(feature)
+
+                # run updated baseline model with the new list of features
+                baseline_model = cross_validate(
+                    self.estimator,
+                    X[self.selected_features_],
+                    y,
+                    cv=self.cv,
+                    scoring=self.scoring,
+                    return_estimator=True,
+                )
```

Comment:
I don't think we need to train this model, because it is the one, we have trained before the if. We only need to replace the baseline_model_performance by model_tmp_performance.

Am I thinking this straight?

Model in line 224 is the same as model 199.